Tests/OptionalTests.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  OptionalTests.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 11/05/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+import CombineExt
+import XCTest
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+final class OptionalTests: XCTestCase {
+    private var subscription: AnyCancellable!
+
+    func testSomeInitialization() {
+        var results = [Int]()
+        var completion: Subscribers.Completion<Never>?
+
+        subscription = Optional(1)
+            .publisher
+            .sink(receiveCompletion: { completion = $0 },
+                  receiveValue: { results.append($0) })
+
+        XCTAssertEqual([1], results)
+        XCTAssertEqual(.finished, completion)
+    }
+
+    func testNoneInitialization() {
+        var results = [Int]()
+        var completion: Subscribers.Completion<Never>?
+
+        subscription = Optional<Int>.none
+            .publisher
+            .sink(receiveCompletion: { completion = $0 },
+                  receiveValue: { results.append($0) })
+
+        XCTAssertTrue(results.isEmpty)
+        XCTAssertEqual(.finished, completion)
```

Comment:
the docs mention this could end with an error, not sure if this is related to nullabiilty or just something confusing about the docs. I imagine it's just gonna compete without anything if it's `nil`?

Reply:
`Optional.Publisher.Failure` is `Never`, so it can’t error. 😄

https://developer.apple.com/documentation/swift/optional/publisher/failure

Reply:
So the docs here are wrong, funny: 
https://developer.apple.com/documentation/swift/optional/publisher/3343960-init